test/e2e_tests/mutating.go

```diff
@@ -0,0 +1,70 @@
+package e2e_tests
+
+import (
+	"context"
+	"fmt"
+
+	cappv1alpha1 "github.com/dana-team/container-app-operator/api/v1alpha1"
+	"github.com/dana-team/container-app-operator/test/e2e_tests/testconsts"
+	mock "github.com/dana-team/rcs-ocm-deployer/test/e2e_tests/mocks"
+	utilst "github.com/dana-team/rcs-ocm-deployer/test/e2e_tests/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	lastUpdatedByAnnotationKey = "rcs.dana.io/last-updated-by"
+	adminAnnotationValue       = "kubernetes-admin"
+)
+
+var _ = Describe("Validate the mutating webhook", func() {
+	It("Should add annotation on create", func() {
+		baseCapp := mock.CreateBaseCapp()
+		baseCapp.Name = utilst.GenerateUniqueCappName(baseCapp.Name)
+		err := k8sClient.Create(context.Background(), baseCapp)
+		Expect(err).NotTo(HaveOccurred())
+
+		fetchedCapp := &cappv1alpha1.Capp{}
+		err = k8sClient.Get(context.Background(), types.NamespacedName{Name: baseCapp.Name, Namespace: baseCapp.Namespace}, fetchedCapp)
+		Expect(err).NotTo(HaveOccurred())
+
+		annotation := fetchedCapp.ObjectMeta.Annotations[lastUpdatedByAnnotationKey]
+		Expect(annotation).To(Equal(adminAnnotationValue))
+	})
+
+	It("Should add annotation on update", func() {
+		baseCapp := mock.CreateBaseCapp()
+		baseCapp.Name = utilst.GenerateUniqueCappName(baseCapp.Name)
+		err := k8sClient.Create(context.Background(), baseCapp)
+		Expect(err).NotTo(HaveOccurred())
+
+		fetchedCapp := &cappv1alpha1.Capp{}
+		err = k8sClient.Get(context.Background(), types.NamespacedName{Name: baseCapp.Name, Namespace: baseCapp.Namespace}, fetchedCapp)
+		Expect(err).NotTo(HaveOccurred())
+
+		annotation := fetchedCapp.ObjectMeta.Annotations[lastUpdatedByAnnotationKey]
+		Expect(annotation).To(Equal(adminAnnotationValue))
+
+		utilst.SwitchUser(&k8sClient, cfg, mock.NSName, newScheme(), true)
+
+		// Use Eventually to retry the update operation until it succeeds
+		Eventually(func() error {
+			fetchedCapp.ObjectMeta.Annotations["test"] = "test"
+			return k8sClient.Update(context.Background(), fetchedCapp)
+		}, testconsts.Timeout, testconsts.Interval).Should(Succeed())
```

Comment:
Once you use the adapter function you probably need need to wrap this in `Eventually`